user/views.py

```diff
@@ -43,7 +54,75 @@ def signup_confirm_view(request):
             return redirect('user:thanks')
     else:
         form = PasswordForm(session_form_data)
-    return render(request, 'user/signupConfirm.html', {'form':form})
+    return render(request, 'user/signup/signupConfirm.html', {'form':form})
 
 def signup_thanks_view(request):
-    return render(request,'user/thanks.html')
+    return render(request, 'user/signup/thanks.html')
+
+# Views for signing in
+class SigninView(LoginView):
+    template_name = 'user/signin.html'
+    next_page = 'user:home'
+
+# Views for signing out
+class SignoutView(LogoutView):
+    template_name = 'user/signedout.html'
+
+# Views for resetting password
+
+class PasswordResetView(PasswordResetView):
+    template_name = "user/password_reset/password_reset_form.html"
+    success_url = reverse_lazy('user:password_reset_done')
+    email_template_name = "user/password_reset/password_reset_email.html"
+
+class PasswordResetDoneView(PasswordResetDoneView):
+    template_name = "user/password_reset/password_reset_done.html"
+
+class PasswordResetConfirmView(PasswordResetConfirmView):
+    success_url = reverse_lazy('user:password_reset_complete')
+    template_name = "user/password_reset/password_reset_confirm.html"
+
+class PasswordResetCompleteView(PasswordResetCompleteView):
+    template_name = "user/password_reset/password_reset_complete.html"
+
+# Views for users
+def user_unauthenticated_view(request):
+    return render(request, 'user/user_unauthenticated.html')
+
+@login_required
+def home_view(request):
+    return render(request, 'user/home.html')
+
+class UserProfileView(LoginRequiredMixin, DetailView):
+    model = CustomUser
+    template_name = 'user/profile/user_profile.html'
+    permission_denied_message = "Oops! Seems like you haven't signed in yet."
+    def get_context_data(self, **kwargs):
+        context = super(UserProfileView, self).get_context_data(**kwargs)
+        page_user = get_object_or_404(CustomUser, id=self.kwargs['pk'])
+        request_user = self.request.user
+        context['request_user']= request_user
+        context['page_user']= page_user
```

Comment:
1. `request_user`についてですが，使用している箇所が見られなかったので削除しましょう
2. メソッドの前に空行を入れましょう
3. イコールの前に空白が抜けています
4. `super()`の引数は，Python3では原則なくてOKです[参考](https://www.lifewithpython.com/2014/01/python-super-function.html)
```suggestion
    permission_denied_message = "Oops! Seems like you haven't signed in yet."

    def get_context_data(self, **kwargs):
        context = super().get_context_data(**kwargs)
        page_user = get_object_or_404(CustomUser, id=self.kwargs['pk'])
        context['page_user'] = page_user
```

Reply:
了解です！少しトピック離れますが、「使用していない」というのがわかるようなvscodeのおすすめのextensionあったりしますか…？

Reply:
これに関しては，左上の虫眼鏡のところから全体検索でrequest_userを検索しました！拡張機能は特に使用していません。
<img width="298" alt="Screen Shot 2022-05-23 at 18 05 14" src="https://user-images.githubusercontent.com/41906969/169784477-8ef0ba6a-80e7-4bdb-979f-f68321f05dc3.png">

---
せっかくなのでトピック外ですがおまけ↓
この件に関しては注意深くチェックしないと気づかないですが，基本的なこと（unused importとか，空行とか，使っていない変数とか）については，flake8による構文解析とautopep8によるフォーマットがおすすめです。一目で気づく+直せるところは勝手に直してくれます。割とこれに関する指摘を多くしている気がするので，導入おすすめ！

VSCodeのsettings.jsonに以下の設定を足しています。
多分`"python.linting.flake8Enabled": true,`と設定に書くと，右下にflake8インストールする？みたいのが出てくるのでOKするとわかるようになるはず
さらに保存するとautopep8入れますか？と出て以降は勝手にフォーマットしてくれるはず
追加で手動でisortを入れると，勝手に並び替えもしてくれます！
（pythonの拡張機能とこれらのpipによるインストールのみでできます）
```json
    "python.linting.pylintEnabled": false,
    "python.linting.flake8Enabled": true,
    "python.linting.flake8Args": [
        "--ignore=W293, E501",
    ],
    "[python]": {
        "editor.defaultFormatter": "ms-python.python",
        "editor.formatOnSave": true,
    },
```

出てきた以下みたいなproblemを減らしといてくれるとgood!
<img width="518" alt="Screen Shot 2022-05-23 at 18 08 20" src="https://user-images.githubusercontent.com/41906969/169785764-bcc321fb-1ade-4418-b7b0-88aa1eb70ec7.png">
<img width="641" alt="Screen Shot 2022-05-23 at 18 08 29" src="https://user-images.githubusercontent.com/41906969/169785779-816c0e62-25ba-46cc-b788-1b0fdb474280.png">